htdocs/core/actions_addupdatedelete.inc.php

```diff
@@ -111,8 +111,10 @@
 			$value = GETPOST($key, 'none');
 		} elseif ($object->fields[$key]['type']=='date') {
 			$value = dol_mktime(12, 0, 0, GETPOST($key.'month'), GETPOST($key.'day'), GETPOST($key.'year'));
-		} elseif ($object->fields[$key]['type']=='datetime') {
-			$value = dol_mktime(GETPOST($key.'hour'), GETPOST($key.'min'), 0, GETPOST($key.'month'), GETPOST($key.'day'), GETPOST($key.'year'));
+        } elseif ($object->fields[$key]['type']=='datetime' ) {
+            $value = dol_mktime(GETPOST($key.'hour'), GETPOST($key.'min'), 0, GETPOST($key.'month'), GETPOST($key.'day'), GETPOST($key.'year'),false,0);
+        } elseif ($object->fields[$key]['type']=='time') {
+            $value = GETPOST($key.'hour')+ GETPOST($key.'min')/60;
```

Comment:
Like for datetime, data of time should be store using the official unit, so the second.
So you should have here $value = GETPOST($key.'hour') * 3600 + GETPOST($key.'min') * 60;
and do the opposite later

Reply:
In business and administration, they are rarely interested in seconds because tasks are executed using hours as units. So representing time in hours has the advantage to be more recognizable when viewing data directly using tools like phpmyadmin.
Furthermore, this is the way they represent time in Odoo.
For this two reasons, I prefer that the first time implementation should be in hours.
This may be changed or extended in the future if we discover that this is not satisfying.

Reply:
You should not make a confusion in what business see and how data are stored.
Time format in php are number of seconds since 1970. Date are also store into memory as seconds (even if this is a full day where we don't mind hours, minutes and seconds). This is just the standard to store data (at least for PHP dates). We should respect this, even if on business side, we show only hour and minutes.